### PR TITLE
Replace randomcolor with hooks

### PR DESCRIPTION
**Pull request by jrmi**

### CODE REVIEW

**Review by CFarcy — APPROVED** [review]

File: src/modules/RA/DataLayer/components/useRandomColor.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { useField } from 'react-final-form';
+import randomColor from 'randomcolor';
+
+/**
+ * Return a random color according with form name as seed.
+ * @param {str} seed
```
Comment (mabhub) [inline src/modules/RA/DataLayer/components/useRandomColor.js:6]:
```suggestion
 * @param {string} seed
```

File: src/modules/RA/DataLayer/components/useSourceData.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { useDataProvider } from 'react-admin';
+import { useField } from 'react-final-form';
+
+import { RES_DATASOURCE } from '../../ra-modules';
+
+/**
+ * Returns the source data
+ * @param {str} sourceFieldName
```
Comment (mabhub) [inline src/modules/RA/DataLayer/components/useSourceData.js:11]:
```suggestion
 * @param {string} sourceFieldName
```

File: src/modules/RA/DataLayer/components/useSourceData.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  const dataProvider = useDataProvider();
+
+  const [sourceData, setSourceData] = React.useState({});
+  const loadingRef = React.useRef(false);
+
+  React.useEffect(() => {
+    let isMounted = true;
+
+    const loadSourceData = async () => {
+      const { data } = await dataProvider.getOne(RES_DATASOURCE, { id: sourceId });
```
Comment (mabhub) [inline src/modules/RA/DataLayer/components/useSourceData.js:24]: Maybe could `dataProvider.getOne()` be replaced by [`useGetOne()`](https://marmelab.com/react-admin/Actions.html#specialized-hooks) specialized hook.

*(Tests should be made to check behaviors precisely)*

File: src/utils/geom.js [hunk 4]
```diff
@@ -20,14 +20,20 @@ export const getShapeFromGeomType = geomType => {
...
   }
 };
 
 export const getLayerStyleDefaultValue = (color, type) => {
   if (!type) {
-    return {};
+    return undefined;
+  }
+  if (type === null) {
+    return null;
```
Comment (dtrucs) [inline src/utils/geom.js:36]: If the `type` parameter equals to `null`, this function should return `undefined`
Maybe you should reverse these two conditions.

File: src/components/react-admin/JSONInput.js [hunk 5]
```diff
@@ -23,14 +25,15 @@ const allowedModes = ['code', 'tree'];
...
-    meta: { error },
+    meta: { touched, error },
   } = useInput(props);
 
   const { source, initialValue = {} } = props;
 
   return (
     <Labeled label={source} {...props}>
       <>
+        {touched && error && (<Typography color="error">{error}</Typography>)}
```
Comment (mabhub) [inline src/components/react-admin/JSONInput.js:36]: Strictly cosmetic:

```suggestion
        {(touched && error) && (
          <Typography color="error">{error}</Typography>
        )}
```

**Review by mabhub — APPROVED** [review]

File: src/modules/RA/DataLayer/components/useSourceData.js (hunk 3 above)
Comment (jrmi) [reply src/modules/RA/DataLayer/components/useSourceData.js:24]: I already tried but I can't do that, a provider can't be conditional ;-)

File: src/utils/geom.js (hunk 4 above)
Comment (jrmi) [reply src/utils/geom.js:36]: No I want undefined in all case. I forget to delete this code.

**Comment by jrmi** [comment]
All modification in next MR...

**Pull request merged**